### PR TITLE
Changes to add params in SPI and SPIXIP Object model

### PR DESCRIPTION
**Pull request by mhtwn**

### CODE REVIEW

File: src/main/scala/devices/spi/OM_SPI.scala [hunk 1]
```diff
@@ -4,7 +4,16 @@ package sifive.blocks.devices.spi
 import freechips.rocketchip.diplomaticobjectmodel.model.{OMDevice, OMInterrupt, OMMemoryRegion}
 
 case class OMSPI(
-  numCS: Int,
+  rxDepth: Int,
+  txDepth:Int,
```
Comment (mwachs5) [inline src/main/scala/devices/spi/OM_SPI.scala]: can you add a space here (manual lint ;)

File: src/main/scala/devices/spi/TLSPIFlash.scala [hunk 2]
```diff
@@ -145,7 +145,10 @@ class TLSPIFlash(w: Int, c: SPIFlashParams)(implicit p: Parameters)
     def getOMComponents(resourceBindings: ResourceBindings, children: Seq[OMComponent] = Nil): Seq[OMComponent] = {
       Seq(
         OMSPIXIP(
-          numCS = c.csWidth,
+          instructionAddressBytes = c.insnAddrBytes,
```
Comment (mwachs5) [inline src/main/scala/devices/spi/TLSPIFlash.scala]: can you check an instance of OMSPIXIP and verify that it has all the parameters that SPI does as well? Its hard for me to tell from this PR how that would Just Work.

Comment (mhtwn) [reply src/main/scala/devices/spi/TLSPIFlash.scala]: No, it doesn't. Is that what we want? The A-spec doesn't say that we want the underlying SPI parameters as well in the Object model for SPI XIP.

File: src/main/scala/devices/spi/OM_SPI.scala (hunk 1 above)
Comment (mhtwn) [reply src/main/scala/devices/spi/OM_SPI.scala]: Sure!

File: src/main/scala/devices/spi/TLSPIFlash.scala (hunk 2 above)
Comment (mwachs5) [reply src/main/scala/devices/spi/TLSPIFlash.scala]: the SPIXIP `extends` SPI so it should similarly extend in the Chisel

Comment (mhtwn) [reply src/main/scala/devices/spi/TLSPIFlash.scala]: you mean in the object model, right?

Comment (mhtwn) [reply src/main/scala/devices/spi/TLSPIFlash.scala]: @mwachs5 It correctly shows the SPI params in SPIXIP now.

**Comment by mhtwn** [comment]
Merging this to allow the other PRs to go through. Changes can be reverted back if needed. Updates (if needed) can also be added as a followup too.

**Pull request merged**